src/lib/mina_networking/mina_networking.ml

```diff
@@ -1438,6 +1438,35 @@ let create (config : Config.t)
   let%map gossip_net =
     Gossip_net.Any.create config.creatable_gossip_net rpc_handlers
   in
+  (* Initialize the ban loop. *)
+  don't_wait_for
+    (Strict_pipe.Reader.iter (Trust_system.ban_pipe config.trust_system)
+       ~f:(fun (banned_peer, expiration) ->
+         let%bind () =
+           let log_failure reason err =
+             [%log error]
+               "Failed to send ban_notify request to $peer we banned (%s)."
+               reason
+               ~metadata:
+                 [ ("peer", `String (Peer.Id.to_string banned_peer.peer_id))
+                 ; ("error", `String (Error.to_string_hum err))
+                 ]
+           in
+           match%map
+             Gossip_net.Any.query_peer gossip_net banned_peer.peer_id
+               Rpcs.Ban_notify expiration
```

Comment:
Are you sure you want `expiration` as timeout here ?

Reply:
`expiration` here is the data to be sent with the RPC to the newly-banned peer. The function [does take a `~timeout` argument](https://github.com/MinaProtocol/mina/blob/e337ff86bfc7d3cb513de1879dc6673e4630f63d/src/lib/gossip_net/libp2p.ml#L716), but it's not given here.

By my reading, this will use the default timeout.

Reply:
@mrmr1993 agree, `expiration` is different.

But also I don't see any default timeout. It's evidenced by line https://github.com/MinaProtocol/mina/blob/e337ff86bfc7d3cb513de1879dc6673e4630f63d/src/lib/gossip_net/libp2p.ml#L607 and what follows it

```match timeout with
                | None ->
                    d
                | Some timeout ->
                    Deferred.choose
                      [ Deferred.choice d Fn.id
                      ; choice (after timeout) (fun () ->
                            Or_error.error_string "rpc timed out")
                      ])```

When no timeout supplied, there will be no timeout at all. Which is bad.

Reply:
Hmm, we should probably update that. No timeout by default on RPCs seems like the wrong way to set this up, and could be hiding bugs/leaks in the code. For the sake of this PR, I'm updating this RPC call to have a 15 second timeout.